cdap-data-fabric/src/main/java/io/cdap/cdap/spi/metadata/DefaultMetadataStorageContext.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2025 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.metadata;
+
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import java.util.Collections;
+import java.util.Map;
+
+// TODO(CDAP-21174): Create metadata storage specify properties
+/**
+ * Default implementation of the {@link MetadataStorageContext}.
+ */
+public class DefaultMetadataStorageContext implements MetadataStorageContext {
+
+    private static final String storageImpl = "gcp-spanner";
```

Comment:
This should not be hardcoded, it should come from provider, see : https://github.com/cdapio/cdap/blob/60e58d46a81a628f2f5b04f1187efbad6b7395c8/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/DefaultCredentialProviderContext.java#L40